code/modules/xenoarcheaology/artifacts/artifact.dm

```diff
@@ -5,25 +5,39 @@
 	icon_state = "ano00"
 	var/icon_num = 0
 	density = 1
+
+	var/predefined_icon_num
+
+	var/datum/component/artifact_master/artifact_master = /datum/component/artifact_master
+
+	var/being_used = 0
+
+/*
 	var/datum/artifact_effect/my_effect
 	var/datum/artifact_effect/secondary_effect
-	var/being_used = 0
 
 	var/predefined_effects = FALSE
 
 	var/predefined_primary
 	var/predefined_secondary
 
-	var/predefined_icon_num
-
 	var/predefined_triggers = FALSE
 
 	var/predefined_trig_primary
 	var/predefined_trig_secondary
+*/
 
 /obj/machinery/artifact/New()
 	..()
 
+	if(ispath(artifact_master))
+		AddComponent(artifact_master)
+
+		artifact_master = GetComponent(artifact_master)
+	else
```

Comment:
I feel like this `else` should be `if(!istype(artifact_master)` since the following lines assume it is an instance of the correct type.